### ISSUE TITLE
IDE0052 fires on write-only properties

### ISSUE THREAD

**Issue by davkean** [dotnet/roslyn#30894]
While write-only properties are questionable, just because they don't have a getter doesn't mean they can be removed.

``` VB
        ''' <summary>
        ''' Sets/Gets the Encoding used for text file resources, without undo support, and without causing the 
        '''  designer to be dirtied (because ComponentChangeService notifications won't get sent)
        ''' </summary>
        ''' <value></value>
        ''' <remarks></remarks>
        Private WriteOnly Property EncodingWithoutUndo() As Encoding
            Set(Value As Encoding)
                Debug.Assert(IsLink AndAlso ResourceTypeEditor.Equals(ResourceTypeEditors.TextFile))
                If _resXDataNode.FileRef IsNot Nothing Then
                    _resXDataNode = NewResXDataNode(Name, Comment, AbsoluteLinkPathAndFileName, ValueTypeName, Value)
                End If
            End Set
        End Property
```

Later: 

``` VB

EncodingWithoutUndo = GuessFileEncoding(AbsoluteLinkPathAndFileName)
```

Expected: No warning
Actual:

```
Severity	Code	Description	Project	Project Rank	File	Line	Category	Suppression State	Tool
Message	IDE0052	Private member 'Resource.EncodingWithoutUndo' can be removed as the value assigned to it is never read.	Microsoft.VisualStudio.Editors	3	E:\project-system2\src\Microsoft.VisualStudio.Editors\ResourceEditor\Resource.vb	958	Code Quality	Active	Compiler
```

**Comment by mavasani** [comment]
This is by design. For majority of realistic code examples, a write-only property is very likely dead code and this violation will help them clean up their code. For questionable examples such as the one mentioned here, the write-only property should be converted to a method OR the violation should be suppressed if it is legacy code.

**Comment by mavasani** [comment]
> Can you show me exactly of what you claim as realistic write-only properties that are dead code?

I don't think there is any modern code base that uses private write-only properties, except if they sneaked in by accident when code was getting refactored. Can you show me any other open source repo, except the legacy VB projects in project-system that uses one?

Regardless, given these are extremely rare, and even if they exist are a questionable API design choice, I would be fine with us detecting and bailing out on these. I am re-opening this issue and will confirm it in the next design meeting.

**Comment by CyrusNajmabadi** [comment]
@mavasani possibly worth having a sub-option for this?  so people can say "actually, in my code we do use this pattern", but others can say "yeah... i would consider that strange, let me get rid of it"

Also, we do have a convert-property-to-method refactoring.  But i'm not actually sure it works on write-only properties.  :D

**Comment by mavasani** [comment]
@CyrusNajmabadi Given these are so rare, for now I am going to submit a PR for this analyzer to bail out on write-only properties to remove the noise that @davkean is seeing.

We should likely have an option OR report a different diagnostic for write-only properties, suggesting replacing them with methods. I will file a separate issue and bring that up for design review next week.

**Comment by mavasani** [comment]
> We should likely have an option OR report a different diagnostic for write-only properties, suggesting replacing them with methods. I will file a separate issue and bring that up for design review next week.

Filed https://github.com/dotnet/roslyn/issues/30983
